strax/processing/general.py

```diff
@@ -126,14 +128,41 @@ def _find_break_i(data, safe_break, not_before):
 
 
 @export
-@numba.jit(nopython=True, nogil=True, cache=True)
 def fully_contained_in(things, containers):
-    """Return array of len(things) with index of interval in containers
+    """
+    Return array of len(things) with index of interval in containers
     for which things are fully contained in a container, or -1 if no such
     exists.
-    We assume all intervals are sorted by time, and b_intervals
-    nonoverlapping.
+    We assume all things and containers are sorted by time.
+    If containers are overlapping, the first container of the thing is chosen.
     """
+    try:
+        _check_time_is_sorted(things['time'])
+    except Exception:
+        raise ValueError('time of things should be sorted!')
+    try:
+        _check_time_is_sorted(containers['time'])
+    except Exception:
+        raise ValueError('time of containers should be sorted!')
+    try:
+        _check_objects_are_not_overlapping(containers)
+    except Exception:
+        warnings.warn(
+            'Overlapping of containers detected! '
+            'fully_contained_in function will only return '
+            'the first container of the thing.')
+    for objects, names in zip([things, containers], ['things', 'containers']):
+        try:
+            _check_objects_non_negative_length(objects)
+        except Exception:
+            raise ValueError(f'{names} should have non-negative length!')
+
+    return _fully_contained_in(things, containers)
+
+
+@numba.jit(nopython=True, nogil=True, cache=True)
+def _fully_contained_in(things, containers):
```

Comment:
Here `_fully_contained_in` become the core function of `fully_contained_in`, in `fully_contained_in`, I added some error message.